File: ros2_ws/ros2_ws/src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewsStationNode: public rclcpp::Node
{
public:
    RobotNewsStationNode(): Node("robot_news_station")
    {
        
        RCLCPP_INFO(this->get_logger(), "Hello Cpp Node)");
    }

private:
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class RobotnewsStationNode(Node):
    def __init__(self):
        super().__init__('robot_news_station')
        self.robot_name_ = "C3PO"
        self.publisher_ = self.create_publisher(String, "robot_news", 10)
        self.timer_ = self.create_timer(0.5, self.publish_news)
        self.get_logger().info("Robot News Sation has been started")


    def publish_news(self):
        msg = String()
        msg.data=f"Hello, this is {self.robot_name_}"
        self.publisher_.publish(msg)



def main(args=None):
    rclpy.init(args=args)
    node = RobotnewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__=="__main__":
    main()